idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/syntaxHighlighter/SyntaxHighlighterSyntax.kt

```diff
@@ -0,0 +1,108 @@
+package arrow.meta.ide.dsl.editor.syntaxHighlighter
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.phases.editor.syntaxHighlighter.SyntaxHighlighterExtensionProvider
+import arrow.meta.phases.ExtensionPhase
+import com.intellij.lang.Language
+import arrow.meta.ide.dsl.editor.color.ColorSyntax
+import com.intellij.lexer.Lexer
+import com.intellij.openapi.editor.colors.TextAttributesKey
+import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory
+import com.intellij.openapi.fileTypes.SyntaxHighlighter
+import com.intellij.openapi.fileTypes.SyntaxHighlighterBase
+import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory
+import com.intellij.psi.tree.IElementType
+import org.jetbrains.kotlin.idea.highlighter.KotlinHighlighter
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.PsiParser
+
+/**
+ * [SyntaxHighlighterExtensionProvider] provides means to register [SyntaxHighlighter]s for Languages.
+ * SyntaxHighlighters solely specify highlighted tokens, which are generated from the Lexer.
+ * [ColorSyntax] allows to highlight additional descriptors from the `Parser` or [Annotator]. You may create a `Parser` with [PsiParser] an example is here [org.jetbrains.kotlin.parsing.KotlinParser].
```

Comment:
I'd remove the note about the parser, as it might be a bit confusing to new users.
Creating a parser isn't helping the user to customize highlighting for an existing file type/language.

The parser is registered when the language is registered (here, by the Kotlin plugin). Even though a parser and annotator use the additional highlighting information, the parser is usually not extendable.

Customization of editor highlighting is usually implemented with an annotator, unless the user is implementing a new parser for a new language.

Reply:
Agree, I think I was too excited, once you send me the Docs about `Annotator` 😄